tests/core/test_broker.py

```diff
@@ -194,25 +193,53 @@ def test_load_non_existent_custom_broker_name():
 
 
 def test_kafka_broker_from_config():
-    endpoints_path = "data/test_endpoints/event_brokers/kafka_plaintext_endpoint.yml"
+    endpoints_path = (
+        "data/test_endpoints/event_brokers/kafka_sasl_plaintext_endpoint.yml"
+    )
     cfg = read_endpoint_config(endpoints_path, "event_broker")
 
     actual = KafkaEventBroker.from_endpoint_config(cfg)
 
     expected = KafkaEventBroker(
         "localhost",
-        "username",
-        "password",
+        sasl_username="username",
+        sasl_password="password",
         topic="topic",
         security_protocol="SASL_PLAINTEXT",
     )
 
-    assert actual.host == expected.host
+    assert actual.url == expected.url
     assert actual.sasl_username == expected.sasl_username
     assert actual.sasl_password == expected.sasl_password
     assert actual.topic == expected.topic
 
 
+@pytest.mark.parametrize(
+    "file,exception",
+    [
+        # `_create_producer()` raises `kafka.errors.NoBrokersAvailable` exception
+        # which means that configuration seems correct but a connection to
+        # broker cannot be established
+        ("kafka_sasl_plaintext_endpoint.yml", kafka.errors.NoBrokersAvailable),
```

Comment:
Optional: parametrize directly over the YAML strings instead of the filenames, and get rid of the files. I think it's a bit cleaner having the content in the test, especially for small files, but really this is up to you